maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultVersionRangeResolver.java

```diff
@@ -111,28 +111,36 @@ public VersionRangeResult resolveVersionRange( RepositorySystemSession session,
         }
         else
         {
-            Map<String, ArtifactRepository> versionIndex = getVersions( session, result, request );
-
-            List<Version> versions = new ArrayList<>();
-            for ( Map.Entry<String, ArtifactRepository> v : versionIndex.entrySet() )
+            if ( Objects.equals( versionConstraint.getRange().getLowerBound(), 
+                                 versionConstraint.getRange().getUpperBound() ) )
+            {
+                result.addVersion( versionConstraint.getRange().getLowerBound().getVersion() );
```

Comment:
why is this shortcut allowed for recommended versions (soft-references)?

Reply:
What happens if the artifact does not exist with the given version ?

Reply:
The same as for soft-requirements, i.e. that would lead to an exception subsequently.